cfpb/cfgov-refresh
# Add back end and test form for submitting comments via Regulations.gov's Comment API

This PR creates the view functions and response templates needed for submitting a comment to the Regulations.gov Comment API.

## Additions

- View functions `regsgov_comment` and `submit_comment`
- Templates for success, user error, and server error responses
- A template with a test form, accessible at `/reg-comment-form-test/`
- Placeholders for necessary environment variables added to `.env_SAMPLE`
- Initial work on tests for the new functions
- Placeholder file for future Wagtail organism, `reg-comment.html` (can be ignored for the purposes of this PR)

## Testing

1. Pull this fork down
2. Copy the new env vars to your `.env` file
3. Replace `<api_key>` with the actual API key (get this from me or @amymok)
4. `cd` out and back into `cfgov-refresh` to activate the new env vars
5. `pip install -r requirements/local.txt`
6. `./cfgov/manage.py runserver`
7. Go to <http://localhost:8000/reg-comment-form-test/>
8. Fill out the form and submit
9. See result page 🎊 

## Review

- @amymok 
- @kave 
- @kurtw 
- @richaagarwal 
- @rosskarchner 

## Todos

- @amymok to finish writing a few tests and making a few other cleanup changes (before merging this PR)
- @Scotchester to finish making organism markup for the actual desired form UI, to be added as a Wagtail module (in another PR)

## Checklist
